tutorials/kubernetes-auth-openid-rbac.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Kubernetes 1.6.1 Authentication via Google OpenID
+description: A sample process how to use Google Accounts with Kubernetes cluster with Role Based Access-Control (RBAC) authorization mode.
+author: miry
+tags: Kubernetes
+date_published: 2017-05-10
+---
+
+In this tutorial you will setup authentication and authorization to your own Kubernetes cluster using your Google account with the help of Role Based Access-Control ([RBAC]) and [OpenID Connect].
+RBAC was introduced in the Kubernetes 1.6 article, RBAC Support in Kubernetes, and was based on Micah Hausle’s [Reduce administrative toil with Kubernetes 1.3](https://www.skuid.com/blog/reduce-administrative-toil-with-kubernetes-1-3/).
+
+### Objectives
+- Creating a Google API Console project and client ID
+- Setup a Kubernetes cluster with kubeadm
+- Generate a local user credentials
+- Grant permissions
+
+## Step 1: Creating a Google API Console project and client ID
+- Go to https://console.developers.google.com/projectselector/apis/library
+- From the project drop-down, select an existing project, or create a new one by selecting **Create a new project**
+- In the sidebar under **API Manager**, select **Credentials**, then select the **OAuth consent screen** tab.
+- Choose an **Email Address**, specify a **Product Name**, and submit **Save**.
+- In the **Credentials** tab, select the **New credentials** drop-down list, and choose **OAuth client ID**.
+- Under **Application type**, select **Other**.
+- From the resulting **OAuth client dialog box**, copy the **Client ID**. The **Client ID** lets your app access enabled Google APIs.
+- Download the client secret JSON file of the credentials.
+
+## Step 2: Setup a Kubernetes cluster
+After initializing the master instance, you need to update the `kube api server` arguments in the `/etc/kubernetes/manifests/kube-apiserver.yaml`. Each argument should be on a separate line. 
+More information about the OIDC attributes can be found in the [Authenticating](https://kubernetes.io/docs/admin/authentication/#option-1---oidc-authenticator) reference documentation.
+
+```
+$ sed -i "/- kube-apiserver/a\    - --oidc-issuer-url=https://accounts.google.com\n    - --oidc-username-claim=email\n    - --oidc-client-id=<Your Google Client ID>" /etc/kubernetes/manifests/kube-apiserver.yaml
+```
+
+Add any network CNI plugin and the cluster is ready. Copy `/etc/kubernetes/admin.conf` to local `~/.kube/config` and change the cluster ip.
+
+```
+$ kubectl get nodes
+NAME            STATUS    AGE       VERSION
+ip-10-9-11-30   Ready     15m       v1.6.1
+```
+
+## Step 3: Generate a local user credentials
+### 3.1 Install the k8s helper on the client machine:
+
+```
+$ go get github.com/micahhausler/k8s-oidc-helper
+```
+
+### 3.2 Generate a user’s credentials for kube config:
+
+```yaml
```

Comment:
looks like this line is misplaced. the paragraph below renders as code block. intentional?